packages/hub/routers.js

```diff
@@ -0,0 +1,280 @@
+const log = require('@cardstack/logger')('cardstack/routers');
+const { declareInjections } = require('@cardstack/di');
+const Error = require('@cardstack/plugin-utils/error');
+const Session = require('@cardstack/plugin-utils/session');
+const defaultRouter = require('@cardstack/routing/cardstack/default-router');
+const { getRoute, isCanonicalRoute } = require('@cardstack/routing/cardstack');
+const { sortBy, uniqBy, get, groupBy } = require('lodash');
+
+const maxRouteDepth = 50;
+const notFoundErrorCardId = 'not-found';
+
+module.exports = declareInjections({
+  currentSchema: 'hub:current-schema',
+  searchers: 'hub:searchers',
+},
+
+class Routers {
+  // Lazily loading this as running this in the constructor runs into the
+  // situation where the searchers are not ready yet. Is there a better way to know when
+  // the searcher is ready so we can set this up outside of a user request?
```

Comment:
I think there's not a better way right now, but we should add one. I [filed a separate issue](https://github.com/cardstack/cardstack/issues/496), because there's already other code like this that should move to the same solution.